ls1intum/Artemis
# [Exam] Exam/management/sessions

### Checklist

### Motivation and Context
![examsession](https://user-images.githubusercontent.com/63286106/85221030-cfca9e80-b3b0-11ea-85b2-ac0b47caa709.png)


### Description
Adds session management logic to start and final view in exam mode

### Steps for Testing
Test locally only
1. Create a few exams directly in the database.
2. Click on the exam in Artemis
3. By default, the Start View will be shown 
4. Check that the Start Exam button is disabled, when the confirmation box is not checked or your correct fullname is not entered.
5. Click on the Start Exam button
5.1. Check in the console whether you have a session token printed there.
5.2. Repeat step 5 a few times, remember your last session token for step 9
6. Change the input boolean of the `exam-participation.component.html` to false in `[startView]=""
7. Click on the exam in Artemis again
8. Now the Final View will be shown
9. Tick the confirmation box and click the Summary button
9.1. Check in the console whether you have a "session token" is the same as "session_token" printed there.
9.2. Make sure that the session token printed here is the same as the last one from step 5

### Screenshots
Start View:
![grafik](https://user-images.githubusercontent.com/63286106/85222934-0f988280-b3bf-11ea-96c5-db171ef12388.png)

Final View:
![grafik](https://user-images.githubusercontent.com/63286106/85222771-1377d500-b3be-11ea-98d4-20a8b18f7ce5.png)
